jwst/outlier_detection/outlier_detection_ifu.py

```diff
@@ -53,203 +49,258 @@ def __init__(self, input_models, reffiles=None, **pars):
+            percentage is based on using all the pixels except a 4 X 4 row and column region around
+            the detector that is often noisy.
+        save_intermediate_results : boolean
+            If True then save intermediate output data
+        ifu_second_check : boolean
+            If True then perform a secondary check searching for outliers. This will set outliers
+            whereever the difference array of adjacent pixels is a Nan. 
+        """
+
+        # set up array to hold group differences
+        diffarr = np.zeros([ndet_files, ny, nx])
+        j = 0
+        for i, model in enumerate(self.input_models):
+            sci = model.data
+            dq = model.dq
+            detector = model.meta.instrument.detector.lower()
+            # only use data from the same detector
+            if detector == uq_det[idet]:
+                bad = np.where(np.bitwise_and(dq, dqflags.pixel['DO_NOT_USE']).astype(bool))
+                # set all science data that have DO_NOT_USE to NAN
+                sci[bad] = np.nan
+
+                # Compute left and right differences (MIRI dispersion axis = 1)
+                # For NIRSpec dispersion axis = 0, these differences are top, bottom
+                # prepend = 0 has the effect of keeping the same shape as sci and
+                # for MIRI data (disp axis = 1) the first column = sci data
+                # OR
+                # for NIRSpec data (disp axis = 0) the first row = sci data
+
+                leftdiff = np.diff(sci, axis=diffaxis, prepend=0)
+                flip = np.flip(sci, axis=diffaxis)
+                rightdiff = np.diff(flip, axis=diffaxis, prepend=0)
+                rightdiff = np.flip(rightdiff, axis=diffaxis)
+
+                # Combine left and right differences with minimum of the abs value
+                # to avoid artifacts from bright edges
+                comb = np.zeros([2, ny, nx])
+                comb[0, :, :] = np.abs(leftdiff)
+                comb[1, :, :] = np.abs(rightdiff)
+                combdiff = np.nanmin(comb, axis=0)
+                diffarr[j, :, :] = combdiff
+                j = j + 1
+
+        # minarr final minimum combined differences, size: ny X nx
+        
+        minarr = np.nanmin(diffarr, axis=0)
+
+        # Normalise the differences to a local median image to deal with ultra-bright sources
+        normarr = medfilt(minarr, kernel_size=kern_size)
+        nfloor = np.nanmedian(minarr)/3
+        normarr[normarr < nfloor] = nfloor  # Ensure we never divide by a tiny number
+        minarr_norm = minarr / normarr
+        # Percentile cut of the central region (cutting out weird detector edge effects)
+        pctmin = np.nanpercentile(minarr_norm[4:ny-4, 4:nx-4], threshold_percent)
+        log.debug("Flag pixels with values above {} {}: ".format(threshold_percent, pctmin))
+        # Flag everything above this percentile value
+        indx = np.where(minarr_norm > pctmin)
```

Comment:
```suggestion
        indx = minarr_norm > pctmin
```

Reply:
@nden @hbushouse The above line gives as different number of pixels being flagged.
I use the len(indx[0]) later in the code  to tell me how many pixels are flagged. I think this is useful information to provide to the user. It could help tell us if something is not working correctly if it flags 10,000 of pixels. Is there another way to find out the number of times minarr_norm> pctmin without using np.where.

Reply:
Ok I used Nadia's trick elsewhere and to get the number of indx = minarr_norm > pctmin I just have to find indx.sum(). So guess that is better than np.where